### PR TITLE
add a function to calculate cell size

### LINKED ISSUE #415: Help: Is there a way to get cellsize
Hello, 

I am new to this package and have been using R before.

In some calculations of grid cells area weighting, I need to extract the grid area. In the previous R package `terra` (`raster`) provides `cellsize` (`area`) function. Is it possible to have a similar implementation in the `rasters` package?

### PR DESCRIPTION
**Pull request by tiemvanderdeure**
Closes #415

### CODE REVIEW

File: ext/RastersArchGDALExt/cellsize.jl [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+function _area_from_coords(transform::ArchGDAL.CoordTransform, ::GI.LinearRingTrait, ring)
+    points = map(GI.getpoint(ring)) do p 
+        t = ArchGDAL.transform!(ArchGDAL.createpoint(p...), transform)
+        (deg2rad(GI.x(t)), deg2rad(GI.y(t)))
+    end
+    return _area_from_rads(GI.LinearRing(points))
+end
+
+function cellsize(x::Tuple{X, Y})
```
Comment (rafaqz) [inline ext/RastersArchGDALExt/cellsize.jl:76]: Its a bit confusing to use `x` for a tuple of dimensions where one is `X`. 

```suggestion
function cellsize(dims::Tuple{X, Y})
```

File: ext/RastersArchGDALExt/cellsize.jl [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    xbnds, ybnds = DimensionalData.intervalbounds(x)
```
Comment (rafaqz) [inline ext/RastersArchGDALExt/cellsize.jl:77]:
```suggestion
    xbnds, ybnds = DimensionalData.intervalbounds(dims)
```

File: ext/RastersArchGDALExt/cellsize.jl [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                    (xb[2], yb[1]), 
+                    (xb[2], yb[2]), 
+                    (xb[1], yb[2]),
+                    (xb[1], yb[1])
+                ]))
+                for xb in xbnds, yb in ybnds]
+        end
+    end
+
+    return Raster(areas, dims = x)
```
Comment (rafaqz) [inline ext/RastersArchGDALExt/cellsize.jl:104]: dims can be the second argument

```suggestion
    return Raster(areas, dims)
```

File: ext/RastersArchGDALExt/cellsize.jl [hunk 4]
```diff
@@ -0,0 +1,109 @@
+"""
+    cellsize(x)
+
+Gives the approximate size of each cell in square km. 
+This function works for any projection, but uses an algorithm for spherical polygons, and thus gives the approximates the true size to about 0.1%, depending on latitude.
```
Comment (rafaqz) [inline ext/RastersArchGDALExt/cellsize.jl:5]:
```suggestion
This function works for any projection, using an algorithm for polygons on a sphere. It approximates the true size to about 0.1%, depending on latitude.
```

File: src/extensions.jl [hunk 5]
```diff
@@ -1,29 +1,22 @@
...
-        if isnothing(Base.get_extension(Rasters, :RastersArchGDALExt))
-            throw(BackendException("ArchGDAL"))
-        else
-            throw(MethodError(resample, args))
-        end
+    if isnothing(Base.get_extension(Rasters, extension))
+        throw(BackendException(package))
     else
-        throw(BackendException("ArchGDAL"))
+        throw(MethodError(resample, args))
```
Comment (rafaqz) [inline src/extensions.jl:7]: Looks like we need to pass the function through as another argument for the methoderror to work

**Comment by tiemvanderdeure** [comment]
Thanks for your comments! The MethodError should work now.

**Review by rafaqz — APPROVED** [review]

**Comment by asinghvi17** [comment]
Just to confirm, this is the Haversine distance, i.e., area by great circle calculation, not area by true geodesic (a la Karney/Vincenty)?

**Comment by tiemvanderdeure** [comment]
It uses Girard's theorem. So it doesn't use Haversine distance exactly, but it is an approximation for spheres.

**Pull request merged**